src/Service/AuthorizationService.php

```diff
@@ -117,33 +117,33 @@ public function isGranted($identity, $permission, $context = null)
         }
 
         if ($this->hasAssertion($permission)) {
-            return $this->assert($this->assertions[(string) $permission], $identity, $context);
+            return $this->assert($this->assertions[(string) $permission], $permission, $identity, $context);
```

Comment:
I see what you mean, but why/when would a permission be an AssertionInterface? I would argue never so the cast can be removed.